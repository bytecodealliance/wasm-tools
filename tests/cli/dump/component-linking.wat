;; RUN: dump %

(component
  (import "a" (instance $i
    (export "a" (core module))
    (export "b" (func))
    (export "c" (value string))
    (export "d" (instance))
    (export "e" (component))
  ))

  (component $c
    (import "a" (core module))
    (import "b" (func))
    (import "c" (value string))
    (import "d" (instance))
    (import "e" (component))
  )

  (instance (instantiate $c
    (with "a" (core module $i "a"))
    (with "b" (func $i "b"))
    (with "c" (value $i "c"))
    (with "d" (instance $i "d"))
    (with "e" (component $i "e"))
  ))
)
