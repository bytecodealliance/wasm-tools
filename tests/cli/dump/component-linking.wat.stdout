  0x0 | 00 61 73 6d | version 12 (Component)
      | 0c 00 01 00
  0x8 | 07 30       | component type section
  0xa | 01          | 1 count
  0xb | 42 09 00 50 | [type 0] Instance([CoreType(Module([])), Export { name: "a", url: "", ty: Module(0) }, Type(Func(ComponentFuncType { params: [], results: Named([]) })), Export { name: "b", url: "", ty: Func(0) }, Export { name: "c", url: "", ty: Value(Primitive(String)) }, Type(Instance([])), Export { name: "d", url: "", ty: Instance(1) }, Type(Component([])), Export { name: "e", url: "", ty: Component(2) }])
      | 00 04 01 61
      | 00 00 11 00
      | 01 40 00 01
      | 00 04 01 62
      | 00 01 00 04
      | 01 63 00 02
      | 73 01 42 00
      | 04 01 64 00
      | 05 01 01 41
      | 00 04 01 65
      | 00 04 02   
 0x3a | 0a 06       | component import section
 0x3c | 01          | 1 count
 0x3d | 01 61 00 05 | [instance 0] ComponentImport { name: "a", url: "", ty: Instance(0) }
      | 00         
 0x42 | 04 59       | [component 0] inline size
   0x44 | 00 61 73 6d | version 12 (Component)
        | 0c 00 01 00
   0x4c | 03 03       | core type section
   0x4e | 01          | 1 count
   0x4f | 50 00       | [core type 0] Module([])
   0x51 | 0a 07       | component import section
   0x53 | 01          | 1 count
   0x54 | 01 61 00 00 | [module 0] ComponentImport { name: "a", url: "", ty: Module(0) }
        | 11 00      
   0x5a | 07 05       | component type section
   0x5c | 01          | 1 count
   0x5d | 40 00 01 00 | [type 0] Func(ComponentFuncType { params: [], results: Named([]) })
   0x61 | 0a 0b       | component import section
   0x63 | 02          | 2 count
   0x64 | 01 62 00 01 | [func 0] ComponentImport { name: "b", url: "", ty: Func(0) }
        | 00         
   0x69 | 01 63 00 02 | [value 0] ComponentImport { name: "c", url: "", ty: Value(Primitive(String)) }
        | 73         
   0x6e | 07 03       | component type section
   0x70 | 01          | 1 count
   0x71 | 42 00       | [type 1] Instance([])
   0x73 | 0a 06       | component import section
   0x75 | 01          | 1 count
   0x76 | 01 64 00 05 | [instance 0] ComponentImport { name: "d", url: "", ty: Instance(1) }
        | 01         
   0x7b | 07 03       | component type section
   0x7d | 01          | 1 count
   0x7e | 41 00       | [type 2] Component([])
   0x80 | 0a 06       | component import section
   0x82 | 01          | 1 count
   0x83 | 01 65 00 04 | [component 0] ComponentImport { name: "e", url: "", ty: Component(2) }
        | 02         
   0x88 | 00 13       | custom section
   0x8a | 0e 63 6f 6d | name: "component-name"
        | 70 6f 6e 65
        | 6e 74 2d 6e
        | 61 6d 65   
   0x99 | 00 02       | component name
   0x9b | 01 63       | "c"
 0x9d | 06 1b       | component alias section
 0x9f | 05          | 5 count
 0xa0 | 00 11 00 00 | alias [module 0] InstanceExport { kind: Module, instance_index: 0, name: "a" }
      | 01 61      
 0xa6 | 01 00 00 01 | alias [func 0] InstanceExport { kind: Func, instance_index: 0, name: "b" }
      | 62         
 0xab | 02 00 00 01 | alias [value 0] InstanceExport { kind: Value, instance_index: 0, name: "c" }
      | 63         
 0xb0 | 05 00 00 01 | alias [instance 1] InstanceExport { kind: Instance, instance_index: 0, name: "d" }
      | 64         
 0xb5 | 04 00 00 01 | alias [component 1] InstanceExport { kind: Component, instance_index: 0, name: "e" }
      | 65         
 0xba | 05 19       | component instance section
 0xbc | 01          | 1 count
 0xbd | 00 00 05 01 | [instance 2] Instantiate { component_index: 0, args: [ComponentInstantiationArg { name: "a", kind: Module, index: 0 }, ComponentInstantiationArg { name: "b", kind: Func, index: 0 }, ComponentInstantiationArg { name: "c", kind: Value, index: 0 }, ComponentInstantiationArg { name: "d", kind: Instance, index: 1 }, ComponentInstantiationArg { name: "e", kind: Component, index: 1 }] }
      | 61 00 11 00
      | 01 62 01 00
      | 01 63 02 00
      | 01 64 05 01
      | 01 65 04 01
 0xd5 | 00 1d       | custom section
 0xd7 | 0e 63 6f 6d | name: "component-name"
      | 70 6f 6e 65
      | 6e 74 2d 6e
      | 61 6d 65   
 0xe6 | 01 05 04    | component section
 0xe9 | 01          | 1 count
 0xea | 00 01 63    | Naming { index: 0, name: "c" }
 0xed | 01 05 05    | instance section
 0xf0 | 01          | 1 count
 0xf1 | 00 01 69    | Naming { index: 0, name: "i" }
